packages/rocketchat-api/server/routes.coffee

```diff
@@ -105,6 +105,69 @@ RocketChat.API.v1.addRoute 'channels.create', authRequired: true,
 		return RocketChat.API.v1.success
 			channel: RocketChat.models.Rooms.findOneById(id.rid)
 
+RocketChat.API.v1.addRoute 'channels.history', authRequired: true,
+	get: ->
+		if not @queryParams.roomId?
+			return RocketChat.API.v1.failure 'Query parameter "roomId" is required.'
+
+		latestDate = new Date
+		if @queryParams.latest?
+			latestDate = new Date(@queryParams.latest)
+
+		oldestDate = undefined
+		if @queryParams.oldest?
+			oldestDate = new Date(@queryParams.oldest)
+
+		inclusive = false
+		if @queryParams.inclusive?
+			inclusive = @queryParams.inclusive
+
+		count = 20
+		if @queryParams.count?
+			count = parseInt @queryParams.count
+
+		unreads = false
+		if @queryParams.unreads?
+			unreads = @queryParams.unreads
+
+		result = {}
+
+		try
+			Meteor.runAsUser this.userId, =>
+				result = Meteor.call 'getChannelHistory', @queryParams.roomId, latestDate, oldestDate, inclusive, count, unreads
+		catch e
+			return RocketChat.API.v1.failure e.name + ': ' + e.message
+
+		return RocketChat.API.v1.success
+			result: result
```

Comment:
@robertsoakes 

1. I agree, good point - we need to have a function which converts the Rocket.Chat message into Slack's format. As it needs to be 100% compatible, that's our aim.
2. That is a good point, we will need to implement the pagination at a higher level - inside of the Meteor methods which are called.